{
  "id": "uno-ooo-parser",
  "version": "0.1.12",
  "timestamp": "2022-01-04 18:58:13+00:00",
  "name": "UseBackupException",
  "type": "exception",
  "namespace": "com.sun.star.embed",
  "parser_args": {
    "sort": true
  },
  "writer_args": {
    "include_desc": true
  },
  "data": {
    "from_imports": [
      [
        "..io.io_exception",
        "IOException"
      ]
    ],
    "from_imports_typing": [],
    "quote": [],
    "typings": [],
    "requires_typing": false,
    "name": "UseBackupException",
    "namespace": "com.sun.star.embed",
    "imports": [],
    "extends": [
      "com.sun.star.io.IOException"
    ],
    "desc": [
      "This exception can be thrown in case a storage commit is failed.",
      "",
      "If a commit process of a storage fails on last transfer and the original content may be corrupted the storage should throw this exception to notify the user that a backup usage is required to restore the original content.",
      "",
      "The storage itself must disconnect from the medium it is based on to allow restoring. Although the storage will still contain all the data internally, and can be used as a temporary storage usually used."
    ],
    "url": "https://api.libreoffice.org/docs/idl/ref/exceptioncom_1_1sun_1_1star_1_1embed_1_1UseBackupException.html",
    "items": {
      "properties": [
        {
          "name": "TemporaryFileURL",
          "returns": "str",
          "desc": [
            "The URL of the temporary file the storage is based on now."
          ],
          "raises_get": "",
          "raises_set": ""
        }
      ]
    }
  }
}